crates/air-lib/trace-handler/src/merger/canon_merger.rs

```diff
@@ -54,26 +50,34 @@ pub(crate) fn try_merge_next_state_as_canon(data_keeper: &mut DataKeeper) -> Mer
 
 fn prepare_both_canon_result(
     prev_canon_result: CanonResult,
-    current_canon_result: &CanonResult,
+    current_canon_result: CanonResult,
 ) -> MergeResult<MergerCanonResult> {
-    check_canon_results(&prev_canon_result, current_canon_result).map_err(MergeError::IncorrectCanonResult)?;
-    prepare_single_canon_result(prev_canon_result)
+    let canon_result =
+        merge_canon_results(prev_canon_result, current_canon_result).map_err(MergeError::IncorrectCanonResult)?;
+    prepare_single_canon_result(canon_result)
 }
 
 fn prepare_single_canon_result(canon_result: CanonResult) -> MergeResult<MergerCanonResult> {
-    Ok(MergerCanonResult::CanonResult(canon_result.0))
+    let merger_result = MergerCanonResult::CanonResult(canon_result);
+    Ok(merger_result)
 }
 
-fn check_canon_results(
-    prev_canon_result: &CanonResult,
-    current_canon_result: &CanonResult,
-) -> Result<(), CanonResultError> {
-    if prev_canon_result != current_canon_result {
-        return Err(CanonResultError::incompatible_state(
-            prev_canon_result.clone(),
-            current_canon_result.clone(),
-        ));
+fn merge_canon_results(
+    prev_canon_result: CanonResult,
+    current_canon_result: CanonResult,
+) -> Result<CanonResult, CanonResultError> {
+    match (&prev_canon_result, &current_canon_result) {
+        (CanonResult::Executed(ref prev), CanonResult::Executed(ref cur)) => {
+            if prev == cur {
+                Ok(prev_canon_result)
+            } else {
+                Err(CanonResultError::incompatible_state(
+                    prev_canon_result,
+                    current_canon_result,
+                ))
+            }
+        }
+        (CanonResult::RequestSentBy(_), CanonResult::Executed(_)) => Ok(current_canon_result),
+        _ => Ok(prev_canon_result),
```

Comment:
Do you think that control flow doesn't get here in case of (Executed, RequestSentBy)? If it does this should be an error or am I wrong?